### PR TITLE
Database Update

### PR DESCRIPTION
**Pull request by izuchukwumba**
This PR contains:
1. Several migrations files showing te progressive iterations the models underwent over time.
2. The database model which includes:
- User, with rows containing user details like username, fullname, imageUrl, skills, interest, groups created etc
- CodeGroup, with rows containing group info like groupName, preferred_category, preferred_availability, preferred_skills etc
- File, with file info like fileName, fileContent, and creator & group info
- Skill and Interest, with associated info like corresponding type, and user user info
- Notification, with details like message, category, senderId, receiverId, groupId, fileId etc

Screenshot 2024-07-16 at 12 33 04 AM
Screenshot 2024-07-16 at 12 33 11 AM
Screenshot 2024-07-16 at 12 33 19 AM
Screenshot 2024-07-16 at 12 33 25 AM
Screenshot 2024-07-16 at 12 33 33 AM
Screenshot 2024-07-16 at 12 33 48 AM
Screenshot 2024-07-16 at 12 34 01 AM

### CODE REVIEW

**Comment by azhiah** [comment]
Were these changes made for the 2nd technical challenge?

**Comment by izuchukwumba** [comment]
Its a general update taking in all your feedback on my previous database setup. Thus, it is made for it, indirectly.

File: frontend/src/app/Nav.jsx [hunk 1]
```diff
@@ -2,6 +2,7 @@ import { useEffect } from "react";
 import { useNavigate } from "react-router-dom";
 import "./Nav.css";
 import axios from "axios";
+// import { notif_categories } from "../notification/notif_categories_frontend";
```
Comment (azhiah) [inline frontend/src/app/Nav.jsx]: Remove commented out code

File: frontend/src/authentication/LoginPage.jsx [hunk 2]
```diff
@@ -98,7 +99,7 @@ function LoginPage() {
       </Button>
       <Box>
         Dont have an account
```
Comment (azhiah) [inline frontend/src/authentication/LoginPage.jsx]: typo

File: frontend/src/authentication/ProfileSetUp.jsx [hunk 3]
```diff
@@ -0,0 +1,192 @@
+import { useState } from "react";
```
Comment (azhiah) [inline frontend/src/authentication/ProfileSetUp.jsx]: I don't see this file used anywhere? Is this related to the database changes? If not it should probably be in a separate PR

**Comment by azhiah** [comment]
> Its a general update taking in all your feedback on my previous database setup. Thus, it is made for it, indirectly.

Do you mean feedback I made in previous PRs?
